### PR TITLE
Log Distribution Fix

### PR DESCRIPTION
**Pull request by AlexCarpenter46**
## Proposed changes

Changes the Interval in Log Distribution for frustums to be the even if the x-coord of the objects aren't the same. This was causing parts of the envelope to not line up properly for unequal mass runs (shown below) which caused bad constraints even at t=0. The second pic is with the fix.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

image
image

### CODE REVIEW

File: src/Domain/CoordinateMaps/Frustum.cpp [hunk 1]
```diff
@@ -130,6 +130,10 @@ Frustum::Frustum(const std::array<std::array<double, 2>, 4>& face_vertices,
     w_plus_ = w_delta + 1.0;
     w_minus_ = w_delta - 1.0;
   } else {
+    zeta_distribution_value_ =
+        distribution_value.has_value()
+            ? distribution_value.value()
+            : -(radius_ + inner_radius_) / (radius_ - inner_radius_);
```
Comment (knelli2) [inline src/Domain/CoordinateMaps/Frustum.cpp]: This can be

```cpp
    zeta_distribution_value_ =
        distribution_value.value_or(
            -(radius_ + inner_radius_) / (radius_ - inner_radius_));
```

**Review by knelli2 — CHANGES_REQUESTED** [review]
Just one thing

File: src/Domain/CoordinateMaps/Frustum.cpp [hunk 2]
```diff
@@ -589,6 +578,9 @@ void Frustum::pup(PUP::er& p) {
     // data from an old version.
     inner_radius_ = std::numeric_limits<double>::signaling_NaN();
   }
+  if (version >= 3) {
+    p | zeta_distribution_value_;
```
Comment (nilsvu) [inline src/Domain/CoordinateMaps/Frustum.cpp:582]: Add an `else` clause which runs when unpacking data written by previous version of the code. It should set `zeta_distribution_value_` to the value use before this change.

File: src/Domain/CoordinateMaps/Frustum.hpp [hunk 3]
```diff
@@ -335,6 +335,7 @@ class Frustum {
   bool with_equiangular_map_{false};
   bool is_identity_{false};
   Distribution zeta_distribution_ = Distribution::Linear;
+  double zeta_distribution_value_{std::numeric_limits<double>::signaling_NaN()};
```
Comment (nilsvu) [inline src/Domain/CoordinateMaps/Frustum.hpp:338]: also add to `==` operator (careful with the `NaN` value)

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Domain/CoordinateMaps/Frustum.hpp (hunk 3 above)
Comment (AlexCarpenter46) [reply src/Domain/CoordinateMaps/Frustum.hpp:338]: When adding to the `==` operator, it seems to break the test in `Test_Domain.cpp` and I'm not exactly sure why. I code --diff'd the 2 domains being compared and they seem to be the same (from the test output). Does the `SerializedDomain.h5` need to be updated?

Comment (nilsvu) [reply src/Domain/CoordinateMaps/Frustum.hpp:338]: I think this is because `inner_radius_` is unpacked to NaN from the data written by the old code. I suggest you set the `zeta_distribution_value_` only if the distribution is set to logarithmic, because only then do you actually need it. (both in the constructor and in pup)

**Comment by nilsdeppe** [comment]
I just want to say nice job on finding and fixing this, @AlexCarpenter46 ! These sorts of things are not always easy to find!

**Comment by knelli2** [comment]
No additional comments

File: src/Domain/CoordinateMaps/Frustum.cpp [hunk 4]
```diff
@@ -612,10 +616,10 @@ bool operator==(const Frustum& lhs, const Frustum& rhs) {
          lhs.half_opening_angle_ == rhs.half_opening_angle_ and
          lhs.one_over_tan_half_opening_angle_ ==
          rhs.one_over_tan_half_opening_angle_ and
-         std::isnan(lhs.zeta_distribution_value_) ==
-             std::isnan(rhs.zeta_distribution_value_) and
-         ((std::isnan(lhs.zeta_distribution_value_) and
-           std::isnan(rhs.zeta_distribution_value_)) or
+         ((std::isnan(lhs.zeta_distribution_value_) or
```
Comment (nilsvu) [inline src/Domain/CoordinateMaps/Frustum.cpp]: why this change? I found the previous easier to understand

**Review by nilsvu — DISMISSED** [review]
Pleas squash, and also run clang-format over the files please

File: src/Domain/CoordinateMaps/Frustum.cpp (hunk 4 above)
Comment (AlexCarpenter46) [reply src/Domain/CoordinateMaps/Frustum.cpp]: Oh for some reason I thought the logic in the other one was causing the difference, so I tried this out. I can revert it back if you want.

Comment (AlexCarpenter46) [reply src/Domain/CoordinateMaps/Frustum.cpp]: Reverted it back

**Review by nilsvu — APPROVED** [review]
Nice 👍

**Review by knelli2 — APPROVED** [review]
Ignoring unrelated test failure
